### ISSUE TITLE
SSH Terminal and Files Tabs not showing on a Linux device on a Local Devices Group (on a LAN mode installation)  

### ISSUE THREAD

**Issue by pkaps** [Ylianst/MeshCentral#3305]
Installed meshcentral in Pop!_OS 21.04 (=ubuntu 20.04) following the directions for ubuntu 18.04 section of the setup guide. Run meshcentral in LAN mode (node ./node_modules/meshcentral). No other fancy software installed, ports 80 and 443 available and allowed to use as per directions, no ufw or other firewalls, (also opensshserver installed (1:8.4p1-5ubuntu1.1), nomachine_7.6.2_4_amd64.deb installed but I do not think it makes a difference) .
node -v :    v12.21.0   
meshcentral -v:  0.9.53, NeDB instance
Had similar behavior like the one mentioned at https://www.youtube.com/watch?v=CHZYFduiOx0
(by Sanusi
2 days ago
Dear Ylian is it already available in version 0.9.52 ? in my lab after adding the local linux devices to the group, I didn't get terminal and files tab. and for local windows machine I don't get web-rdp link... thx)

Had similar behavior running 2 instances of meshcentral as docker instances on an UnRaid server (one instance is NeDB and the other is MongoDB)

Here is a screenshoot from the Pop!_OS instance (=non docker...)

![image](https://user-images.githubusercontent.com/376705/143670338-de947c75-d8c6-43e9-8980-7e02e9cbc471.png)
![image](https://user-images.githubusercontent.com/376705/143670382-51b72bb7-9489-44e4-9ac5-8312e4e87067.png)

**Comment by pkaps** [comment]
Forgot to mention.. 
If I install the agent on the Device/Client machine everything works as expected. I can see all the tabs properly and connect to the device/client, browse their files via the Files Tab etc.

**Comment by Ylianst** [comment]
Hi. Can you share your config.json file? Please replace any confidential information with XXXXX. I will create a server with the same configuration and debug it. Thanks!

**Comment by pkaps** [comment]
Hi Ylian,

I am attaching all 3 files for the 3 different installations.
The POP!_OS one seems to have just used a sample config.json.
Maybe it is my (a user) fault and I just need to properly configure...

Regards,

Panos

**Comment by Ylianst** [comment]
Ok, you are 100% right. Looking at the problem and will work on a fix now!

**Comment by Ylianst** [comment]
Ha, I was a bit of a dummy. I now realize that I forgot to mention that 'mstsc' and 'ssh' need to be set to true in the domain section of the config.json like this:

```
{
  "settings": {
    "port": 443
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "mstsc": true,
      "ssh": true
    }
  }
}
```

Enabling `ssh` will make the terminal and files tab show up. Enabling `mstsc` will make the Web-RDP link show up.

I am starting to think that to lease for SSH, I should enable it by default since it's so useful.

**Comment by pkaps** [comment]
Hello Ylian,

Thank you for your prompt response looking into this.
After adding the above 2 entries to the config.json files, I confirm that
now both the regular and docker instances work as expected for SSH and the
Files tab.
I can connect and browse, download files etc.

Windows RDP still has issues.

I get the Web-RDP for a windows client on the Local Devices Group (the one
without the agent) as you mentioned in your above comment.
[image: image.png]
When clicking A new window for entering the credentials opens as expected:
[image: image.png]
The machine I am trying to connect has RDP enabled and I can connect as
expected to it from another windows machine with the regular RDP windows
client when using the same credentials I am entering here.This is a virtual
machine although it shouldn't matter.
Tried another one too (non virtual machine - regular windows host) with
no luck.
Tried both a local user and microsoft account - the 2 machines have
different users...
I tried all the 192.168.23.8\xxxxx  ***@***.*** combinations for the
credentials with no luck.
Tried both on chrome and firefox.

So basically I conclude that most probably the web-RDP does not work.
Do I need to install an RDP chrome/firefox extension or does meshcentral
take care of it?
Do you think it is a NTLM security issue?

On the same meshcentral instance another (3rd...) windows machine with
meshcentral agent installed works as expected:

[image: image.png]

Thanks again for your help and the great software you have built.

Regards,

Panos

**Comment by si458** [comment]
this has been fixed, please use the latest release 1.1.18
create 'local device, no agent', add device (set as windows (rdp), new desktop tab is shown with 'RDP connect' button
